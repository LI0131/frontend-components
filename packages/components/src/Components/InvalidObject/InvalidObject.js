import React from 'react';
import PropTypes from 'prop-types';
import Icon404 from './icon-404';
import {
    Title,
    Button
} from '@patternfly/react-core';
import { withRouter } from 'react-router-dom';

const InvalidObject = (props) => {
    return (
        <section className="pf-l-page__main-section pf-c-page__main-section ins-c-component_invalid-componet">
            <Title size='3xl'>This doesn&apos;t seem to exist.</Title>
            <Icon404/>
            <Title size='xl' className='ins-c-text__sorry'>Sorry, we couldn&apos;t find what you&apos;re looking for.
                    The page you requested may have changed or moved.</Title>
            <Button
                variant="link"
                onClick={ () => { props.history.goBack(); } }>
                    Go Back
            </Button>
        </section>
    );
};

InvalidObject.propTypes = {
    history: PropTypes.shape({
        goBack: PropTypes.func
    })
};

InvalidObject.defaultProps = {
    history: {
        push: () => undefined,
        goBack: () => undefined
    }
};

export default withRouter(InvalidObject);
