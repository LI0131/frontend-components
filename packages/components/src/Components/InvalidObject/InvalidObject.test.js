import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import InvalidObject from './InvalidObject';

describe('InvalidObject component', () => {
    it('should render', () => {
        const wrapper = shallow(<InvalidObject/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
