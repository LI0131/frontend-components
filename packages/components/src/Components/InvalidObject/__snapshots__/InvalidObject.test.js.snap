// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidObject component should render 1`] = `
<Route>
  <Component />
</Route>
`;
